src/Components/TagPickerGridComponent/TagPickerGridComponent/TagPickerBaseComponent.ts

```diff
@@ -2,247 +2,295 @@ import * as React from 'react';
-	 * @param filter Text used to filter suggestions.
-	 */
-	private searchTags(filter?: string): Promise<ITag[]> {
-		let options = `?$select=${this.idAttribute},${this.nameAttribute}&$orderby=${this.nameAttribute} asc`;
-
-		if (filter)
-			options = `${options}&$filter=contains(${this.nameAttribute},'${filter}')`;
-
-		return this.context.webAPI.retrieveMultipleRecords(this.relatedEntity, options).then(
-			results => {
-				if (results.entities.length < 1)
-					return [];
-
-				return results.entities.map(item => ({ key: item[this.idAttribute], name: item[this.nameAttribute] }));
-			}
-		);
-	}
-
-	/**
-	 * A callback for when the selected list of items changes.
-	 * @param items A collection containing the items.
-	 */
-	private onChange(items?: ITag[]) : void {
-		const promises: Promise<Response>[] = [];
-		const entityExists: boolean = (this.entityId !== undefined && this.entityId !== "00000000-0000-0000-0000-000000000000");
-
-		// We only need to associate / dissasociate items when the entity exists.
-		if (entityExists)
-		{
-			// Associate the added items.
-			const itemsAdded = items?.filter(item => !this.selectedItems.some(selectedItem => selectedItem.key === item.key)) || [];
-			for(let item of itemsAdded) {
-				promises.push(this.associateItem(item));
-			}
-
-			// Disassociate the removed items.
-			const itemsRemoved = this.selectedItems.filter(selectedItem => !items?.some(item => item.key === selectedItem.key));
-			for (let item of itemsRemoved) {
-				promises.push(this.disassociateItem(item));
-			}
-		}
-
-		Promise.all(promises).then(
-			results => {
-				this.props.selectedItems = this.selectedItems = items || [];
-
-				if (!entityExists)
-					this.notifyOutputChanged();
-			}
-		);
+     * Retrieves the label based on the current language.
+     */
+    private getLabelText(): string {
+        if (this.labelText.indexOf("|") === -1)
+            return this.labelText;
+
+        // load the translations from the label
+        const translations = new Map<Languages, string>();
+        for (const translation of this.labelText.split("|")) {
+            const values: string[] = translation.split("=");
+
+            // check if the string is formatted correctly
+            if (values.length !== 2)
+                continue;
+
+            const language: Languages = (<any>Languages)[values[0]];
+
+            // check if the language is supported
+            if (language === undefined)
+                continue;
+
+            const label: string = values[1];
+            translations.set(language, label);
+        }
+
+        const languageId: number = (<any>this.context).userSettings.languageId;
+        const userLanguage: Languages = (<any>Languages)[languageId];
+
+        // check if the user language is supported, otherwise default to English
+        const labelLanguage: number = userLanguage ? languageId : Languages.en;
+
+        return translations.get(labelLanguage) ?? "";
```

Comment:
this becomes slightly more confusing than intended with all the casting. If there's a way to make it clearer that would be great. Either way, at least adding expected format of the string would be helpful, perhaps also throw that in the an usage readme.

Reply:
Updated README accordingly